### PR TITLE
Thread lib: #44 Missing binary semaphore in the common library

### PR DESCRIPTION
**Pull request by forGGe**
Semaphore implemented and simple cleanup is made

/cc @vadimol @GamovCoder

### CODE REVIEW

**Comment by vadimol** [comment]
Looks good for me

File: lib/thread/common/export/ecl/thread/common/mutex.hpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace common
 class mutex
 {
 public:
-    constexpr mutex();
+    constexpr mutex() { }
```
Comment (GamovCoder) [inline lib/thread/common/export/ecl/thread/common/mutex.hpp]: please use `= default` to make emphasis that default construction is what you want. It'll also be inlined.

File: lib/thread/common/semaphore.cpp [hunk 2]
```diff
@@ -27,3 +27,24 @@ ecl::err ecl::common::semaphore::try_wait()
...
+//------------------------------------------------------------------------------
+
+void ecl::common::binary_semaphore::signal()
+{
+    m_flag = true;
+}
+
+void ecl::common::binary_semaphore::wait()
+{
+    while (!m_flag) { }
```
Comment (GamovCoder) [inline lib/thread/common/semaphore.cpp]: Are braces really needed?

File: lib/thread/common/export/ecl/thread/common/mutex.hpp (hunk 1 above)
Comment (forGGe) [inline lib/thread/common/export/ecl/thread/common/mutex.hpp]: Done

File: lib/thread/common/semaphore.cpp (hunk 2 above)
Comment (forGGe) [inline lib/thread/common/semaphore.cpp]: Done

**Pull request merged**